JMdictFiles/sameKanjiCounter.js

```diff
@@ -0,0 +1,51 @@
+const { JM } = require('./dic');
+
+const createKanjiToCountObject = (dict) => {
+    const kanjiToCount = {};
+    for (let i = 0; i < dict.length; i++) {
+        kanjiToCount[dict[i].kanji] = (kanjiToCount[dict[i].kanji] || 0) + 1;
+    }
+    return kanjiToCount;
+};
+
+const kanjiToCountWithAllKanjis = createKanjiToCountObject(JM);
+
+const createNonDistinctKanjiToCountObject = () => {
+    const nonUniqueKanjiToCount = {};
+    for (const kanji in kanjiToCountWithAllKanjis) {
```

Comment:
instead of using a variable defined outside of the scope of the function, please pass the required data as arguments to the function.

Functions should be black boxes: 
```
---input---->[function]---output-->
```
if the function depends on stuff defined outside, you will have problems testing it in isolation - it is also easy to lost track of code executions and what is the actual state of the application (when your code grows).

If you find it convenient to have some local state shared among a couple functions, write a class instead.